source/solvers/heat_transfer.cc

```diff
@@ -648,22 +637,83 @@ HeatTransfer<dim>::postprocess(bool first_iteration)
         }
     }
 
-  // Minimum and maximum temperature
-  if (simulation_parameters.post_processing.calculate_min_max_temperature)
+  // Set-up domain name for output files
+  Parameters::FluidIndicator monitored_fluid =
+    this->simulation_parameters.post_processing.postprocessed_fluid;
+  // default: monophase simulations
+  std::string domain_name("all_domain");
+  bool        gather_vof(false);
+
+  if (this->simulation_parameters.physical_properties_manager
+        .get_number_of_fluids() == 2)
+    {
+      // Multiphase flow
+      gather_vof = true;
+      switch (monitored_fluid)
+        {
+          default:
+            {
+              domain_name = "all_domain";
+              break;
+            }
+          case Parameters::FluidIndicator::fluid0:
+            {
+              domain_name = "fluid_0";
+              break;
+            }
+          case Parameters::FluidIndicator::fluid1:
+            {
+              domain_name = "fluid_1";
```

Comment:
I am not sure the underspaces are required other than if you use these for the output file name